File: rabbitmq-python/producer.py
import pika

connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
channel = connection.channel()

queue_name = 'hola'
message = 'MENSAJE DE PYTHON'

channel.queue_declare(queue=queue_name, durable=True)  # Establece durable en True
channel.basic_publish(exchange='', routing_key=queue_name, body=message, properties=pika.BasicProperties(delivery_mode=2))  # Establece delivery_mode en 2

print(f"[x] Sent '{message}'")

connection.close()

File: rabbitmq-nodejs/producer.js
const amqp = require('amqplib');

async function main() {
    const connection = await amqp.connect('amqp://localhost');
    const channel = await connection.createChannel();

    const queueName = 'hola';
    const message = 'Mensaje de Javascript';

    await channel.assertQueue(queueName, { durable: true });
    await channel.sendToQueue(queueName, Buffer.from(message), { persistent: true });

    console.log([x] Sent '${message}');

    await channel.close();
    await connection.close();
}

main().catch(console.error);
